src/common/utils/currencyConvert.ts

```diff
@@ -0,0 +1,101 @@
+/**
+ * Highly inspired by: https://github.com/AryanJ-NYC/bitcoin-conversion
+ */
+import axios from "axios";
+import currencyJs from "currency.js";
+import { getSettings } from "~/common/lib/api";
+import { SupportedCurrencies } from "~/types";
+
+const settings = async () => {
+  const { currency, exchange } = await getSettings();
+
+  return {
+    currency,
+    exchange,
+  };
+};
+
+const numSatsInBtc = 100_000_000;
+
+export const getBalances = async (balance: number) => {
+  const { currency } = await settings();
+  const fiatValue = await satoshisToFiat({
+    amountInSats: balance,
+    convertTo: currency,
+  });
+  const localeFiatValue = fiatValue.toLocaleString("en", {
+    style: "currency",
+    currency: currency,
+  });
+  return {
+    satsBalance: `${balance} sats`,
+    fiatBalance: localeFiatValue,
+  };
+};
+
+const getFiatBtcRate = async (
```

Comment:
can we cache this? right now a request is made every time an input number is changed. 

wondering also if this should be done in the background script and the rate should be stored in the settings storage?

Reply:
Tried to address this here: https://github.com/getAlby/lightning-browser-extension/pull/1000/files#diff-8532ac3fc9044b3365001d3ea67200d23b63d5edfd398f7d772954117e92e11aR56

But I guess you're right. Instead of `€36,139.34` we could write `~€36,139.34` and only get the _exact_ rate every i.e. x minutes.  
Easiest way I could imagine is setting a long debounce (i.e. 1 hour) on this function.

The background idea might work as well. Not sure how we would throttle the calls. Maybe just the same way?  
Let's talk?